app/core/store/modules/prepaids.js

```diff
@@ -269,66 +270,74 @@ export default {
+        .map(data => data instanceof User ? data : new User(data))
+        .filter(u => u.isEnrolled() && u.prepaidType() === 'course')
 
       const existsLicenseToRevoke = students.length > 0
       if (!existsLicenseToRevoke) {
         noty({ text: `No licenses applied to selected student(s).` })
         return
       }
 
-      let confirmed = false
-      await new Promise((resolve) => noty({
-        text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
-        buttons: [
-          {
-            addClass: 'btn btn-primary',
-            text: 'Ok',
-            onClick: function ($noty) {
-              confirmed = true
-              $noty.close()
-              resolve()
-            }
-          },
-          {
-            addClass: 'btn btn-danger',
-            text: 'Cancel',
-            onClick: function ($noty) {
-              $noty.close()
-              resolve()
+      if (!confirmed) {
+        await new Promise((resolve) => noty({
+          text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
+          buttons: [
+            {
+              addClass: 'btn btn-primary',
+              text: 'Ok',
+              onClick: function ($noty) {
+                confirmed = true
+                $noty.close()
+                resolve()
+              }
+            },
+            {
+              addClass: 'btn btn-danger',
+              text: 'Cancel',
+              onClick: function ($noty) {
+                $noty.close()
+                resolve()
+              }
             }
-          }
-        ]
-      }))
+          ]
+        }))
 
-      if (!confirmed) {
-        return
+        if (!confirmed) {
+          return
+        }
       }
 
-      const promises = []
       for (const student of students) {
         const courseProducts = student.activeProducts('course')
-        courseProducts.map(product => {
-          let prepaid = new Prepaid({
+        await Bluebird.map(courseProducts, async product => {
+          const prepaid = new Prepaid({
             _id: product.prepaid,
             type: 'course'
           })
-          promises.push(new Promise((resolve, reject) =>
+          await new Promise((resolve, reject) =>
             prepaid.revoke(student, {
               success: resolve,
-              error: () => {
-                console.error(`Didn't revoke this license`)
-                resolve()
-              },
+              error: reject,
               data: { sharedClassroomId }
             })
-          ))
+          )
+
+          if (updateUserProducts) {
+            student.set(
+              'products',
+              student.get('products').map((p) => {
```

Comment:
not sure products will be an array for all users especially old ones so would be a good idea to do `(student.get('products') || [])`